android/src/main/java/com/reactcommunity/rndatetimepicker/RNDatePickerDialogFragment.java

```diff
@@ -86,6 +87,10 @@ DatePickerDialog getDialog(
                   display
           );
         default:
+          if (args != null && args.containsKey(RNConstants.ARG_LOCALE)) {
+            // for header (NOTE: will change the app locale)
+            Locale.setDefault(getLocale(args));
```

Comment:
It's quite possible there is no other way to localize the pickers, but this seems like an action that can have far-reaching consequences. 

on [ios](https://developer.apple.com/documentation/uikit/uidatepicker#2281467) the prop just "overrides the system default with a specific locale." - that is a "safe way" of specifying locale.

Here, the call would have side effects beyond the datepicker, and I'm not sure we want to take that path. Seem to me that code like this should probably be called from userland upon app start. What do you think?


(also applies to `activityContext.getResources().getConfiguration().setLocale(getLocale(args))`).

Reply:
I know and I do understand your point. I was looking for a way to localize it without localizing the entire app but couldn't find another way. I was also thinking that passing the 'locale' prop to localize the picker is made for an already localized app when the user what to have the picker in the app's locale, not the device's. Thus no real harm is done when we change the locale this way. We can add a note to these side-effects when using this feature. What do you think?

Reply:
I would prefer to not do it this way. I'm not even sure setting locale is a job for this module, maybe it should be done in the scope of some different package..

If we're to do this, I would prefer for this locale-changing logic to be exposed via a `@ReactMethod` that users can call. That will express the side effect more clearly.